### PR TITLE
SRVCOM-1420 Test Serverless teardown after upgrade

### PR DESCRIPTION
**Pull request by mgencur**
* including CRD deletion

Related to https://issues.redhat.com/browse/SRVCOM-1420

### CODE REVIEW

File: hack/lib/serverless.bash [hunk 1]
```diff
@@ -358,6 +358,10 @@ function teardown_serverless {
   done
   oc delete namespace openshift-serverless --ignore-not-found=true
 
+  if [[ ! $(oc get crd -oname | grep -c 'knative.dev') -eq 0 ]]; then
+    oc get crd -oname | grep 'knative.dev' | xargs oc delete --timeout=60s
+  fi
```
Comment (markusthoemmes) [inline hack/lib/serverless.bash:363]: Are we expecting stuff to hang/fail if it doesn't work as we expect?

**Review by markusthoemmes — COMMENTED** [review]
Looks sound, are we already testing shutdown in non-upgrade paths?

File: hack/lib/serverless.bash (hunk 1 above)
Comment (mgencur) [reply hack/lib/serverless.bash:363]: If the deletion of CRDs hangs it means there's a problem, but hanging is limited to 60 seconds. I didn't want the whole job to be timing out after 2hrs.

Regarding running shutdown in non-upgrade paths. The serverless-operator Golang tests deploy/undeploy KnativeServing and the operator but they don't delete the CRDs. I'm not aware of other places where we'd be testing that.

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

**Pull request merged**